addon/components/service-worker-update-notify.js

```diff
@@ -1,23 +1,57 @@
+import Ember from 'ember';
 import Component from '@ember/component';
 import { isEmpty } from '@ember/utils';
 import { task, timeout } from 'ember-concurrency';
+import withTestWaiter from 'ember-concurrency-test-waiter/with-test-waiter';
 
-import layout from '../templates/components/service-worker-update-notify'
-import serviceWorkerHasUpdate from '../utils/service-worker-has-update'
+import layout from '../templates/components/service-worker-update-notify';
+import serviceWorkerHasUpdate from '../utils/service-worker-has-update';
+
+
+async function update() {
+  const reg = await navigator.serviceWorker.register(
+    '{{ROOT_URL}}{{SERVICE_WORKER_FILENAME}}',
+    { scope: '{{ROOT_URL}}' }
+  );
+
+  reg.update();
+}
 
 export default Component.extend({
   layout,
-  pollingInterval: 1200000, // 20 minutes in ms
-
   tagName: '',
 
+  // private
   hasUpdate: false,
 
+  // public
+  pollingInterval: 1200000, // 20 minutes in ms
```

Comment:
It would be awesome if the polling interval could be configured via `config/environment.js`. That value may typically depend on the environment. Beside the `1ms` case for `test` one might use a more frequent polling on staging than on production to help with testing.

Reply:
How would you add a test that covers that scenario?

Right now, I'm avoiding testing polling, but for no reason, other than I don't know how atm

Reply:
You can set a polling interval in `config/environment.js` and read it in your component and pass it down to the notify component.

Reply:
I mean for the addon's tests

Reply:
> How would you add a test that covers that scenario?

Testing that the `pollingInterval` is actually used? I think [sinon's fake timers](https://sinonjs.org/releases/v7.5.0/fake-timers/) would help here.

Reply:
> You can set a polling interval in `config/environment.js` and read it in your component and pass it down to the notify component.

Sure, that's my work-a-round but actually adds a lot of complexity for a common use case.

Reply:
ok, so, before this is merged, I need at least two tests around the `pollingInterval` arg

Reply:
for config/environment, I'd like to do that in a separate PR, so this one doesn't get too big

Reply:
https://github.com/topaxi/ember-service-worker-update-notify/pull/18